src/java.base/share/classes/jdk/internal/foreign/layout/AbstractLayout.java

```diff
@@ -140,17 +133,17 @@ String decorateLayoutString(String s) {
             s = String.format("%s(%s)", s, name().get());
         }
         if (!hasNaturalAlignment()) {
-            s = bitAlignment + "%" + s;
+            s = bitAlignment() + "%" + s;
         }
         return s;
     }
 
-    private static void checkAlignment(long alignmentBitCount) {
-        if (((alignmentBitCount & (alignmentBitCount - 1)) != 0L) || //alignment must be a power of two
-                (alignmentBitCount < 8)) { //alignment must be greater than 8
-            throw new IllegalArgumentException("Invalid alignment: " + alignmentBitCount);
+    private static long requirePowerOfTwoAndGreaterOrEqualToEight(long value) {
```

Comment:
Should this stay here, or be merged inside MemoryLayoutUtil? It seems the same as `requireBitSizeValid`, except for the zero behavior?

Reply:
This method also checks if the parameter is an even power of two.